extensions/blocks/contact-form/components/jetpack-field-checkbox.js

```diff
@@ -2,31 +2,45 @@
  * External dependencies
  */
 import { __ } from '@wordpress/i18n';
-import { BaseControl, PanelBody, TextControl, ToggleControl } from '@wordpress/components';
-import { Fragment } from '@wordpress/element';
+import { BaseControl, PanelBody, ToggleControl } from '@wordpress/components';
+import { useEffect } from '@wordpress/element';
 import { InspectorControls } from '@wordpress/block-editor';
-import { withInstanceId } from '@wordpress/compose';
+import { compose, withInstanceId } from '@wordpress/compose';
+import { withSelect } from '@wordpress/data';
 
 /**
  * Internal dependencies
  */
 import JetpackFieldLabel from './jetpack-field-label';
 
-const JetpackFieldCheckbox = ( {
-	instanceId,
-	required,
-	label,
-	setAttributes,
-	isSelected,
-	defaultValue,
-	id,
-} ) => {
+function JetpackFieldCheckbox( props ) {
+	const {
+		instanceId,
+		required,
+		label,
+		setAttributes,
+		isSelected,
+		defaultValue,
+		parentBlock,
+		spacing,
+	} = props;
+
+	useEffect( () => {
+		if ( parentBlock && parentBlock.attributes.spacing !== spacing ) {
+			setAttributes( { spacing: parentBlock.attributes.spacing } );
+		}
+	} );
```

Comment:
I think we could optimize this by only triggering if relevant variables change. E.g.
```js
// We could also move this in the withSelect, since it's the only thing we need from parentBlock
const parentBlockSpacing = get( parentBlock, [ 'attributes', 'spacing' ] );

useEffect( () => {
  if ( parentBlockSpacing ) {
    setAttributes( { spacing: parentBlockSpacing } );
  }
}, [ parentBlockSpacing, setAttributes, spacing ] );
```

Reply:
Thanks, that is much better. I think I'm actually going to take all of this code out for now until these options can be supported on the front end.

Reply:
Thanks this is much better. I'm actually going to remove all of this code until a time when the front end code can be updated to support it. I think it's going to hold up this PR too much at this point.